tools/codegen/gen.py

```diff
@@ -334,10 +334,19 @@ def gen_one(f: NativeFunction) -> Optional[str]:
 """
 
             elif self.target is Target.REGISTRATION:
+                dispatcher_sig = DispatcherSignature.from_schema(f.func)
+
                 if local.use_c10_dispatcher() is UseC10Dispatcher.full:
-                    payload = f'TORCH_FN({sig.name()})'
+                    payload = f"TORCH_FN({sig.name()})"
+                elif local.use_c10_dispatcher() is UseC10Dispatcher.hacky_wrapper_for_legacy_signatures:
+                    payload = f"""
+c10::impl::hacky_wrapper_for_legacy_signatures<
+    {dispatcher_sig.type()},
+    std::index_sequence<{', '.join(mutable_argument_indices(f.func))}>>(TORCH_FN({sig.name()}))
+"""
```

Comment:
Should be easy to avoid the copy pasted payload here and on 437 I hope - a little helper function rather than a CodeTemplate ideally

Reply:
I'm following the style in the rest of the file where we duplicate and explicitly spell out generated code instead of putting it into subroutines, search for example for "findSchemaOrThrow". But I'm open to writing it either way, wdyt?

Reply:
Actually, this is a misunderstanding of why `gen_structured` is copy-pasted from `gen_unstructured`: the goal is that structured kernels never have any of the legacy crud that is in unstructured.

Of course, the reality is more bleak; gen structured has to test for `use_c10_dispatcher: full` because it prior to this patch series, it literally does not work. But I feel like at least for `hacky_wrapper_for_legacy_signatures` we should be able to avoid this for structured kernels, right? :o)

Reply:
Yes. There's only two ops that are structured and I can volunteer to do the port. Need to take a look.